internal/pkg/coordinator/monitor_integration_test.go

```diff
@@ -347,7 +347,7 @@ func TestMonitorUnenrollerSetAndClear(t *testing.T) {
 	assert.True(t, agent.Active)
 	// Make sure canceller is no longer there
 	assert.Len(t, pm.(*monitorT).policies, 1)
-	assert.Len(t, pm.(*monitorT).policiesCanceller, 0)
+	assert.Equal(t, pm.(*monitorT).ActivePoliciesCancellerCount(), 0)
```

Comment:
I attest this is a bit ugly, but the code is structured in a way where internal state is both accessed in the test suite and also on goroutine. I've created a specific issues to cover some of the problem https://github.com/elastic/fleet-server/issues/1744

I've took some time last week to create a unit test that could possibly trigger that, but I think this is more a problem in the way the code is structured and how it manages the internal goroutine.